### PR TITLE
Mount user specific secrets in /config

### PR DESCRIPTION
**Pull request by sdake**
Copy the user specific secrets to the correct location and change
 their ownership/permissions to improve usability across a variety
 of platforms.

### CODE REVIEW

File: docker/build-tools/Dockerfile [hunk 1]
```diff
@@ -154,6 +155,13 @@ ADD https://github.com/GoogleCloudPlatform/docker-credential-gcr/releases/downlo
 RUN tar -xzf /tmp/docker-credential-gcr_linux_amd64-${GCR_AUTH_VERSION}.tar.gz -C /tmp
 RUN mv /tmp/docker-credential-gcr ${OUTDIR}/usr/bin
 
+# Install su-exec which is a tool that operates like sudo without the overhead
+ADD https://github.com/ncopa/su-exec/archive/v0.2.tar.gz /tmp
```
Comment (geeknoid) [inline docker/build-tools/Dockerfile]: Use SU_EXEC_VERSION

File: docker/build-tools/docker-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Copy credentials from mountpoints using su-exec
+
+uid = $(id -u)
```
Comment (fpesce) [inline docker/build-tools/docker-entrypoint.sh]: To make linter happy

```suggestion
uid=$(id -u)
```

File: docker/build-tools/docker-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+gid = $(gid -u)
```
Comment (fpesce) [inline docker/build-tools/docker-entrypoint.sh]: idem
```suggestion
gid=$(gid -u)
```

File: docker/build-tools/docker-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+su-exec 0:0 cp -R /mnt/config/* /home
+su-exec 0:0 chown -R ${uid}:${gid} /home
```
Comment (fpesce) [inline docker/build-tools/docker-entrypoint.sh]:
```suggestion
su-exec 0:0 chown -R "${uid}":"${gid}" /home
```

File: docker/build-tools/docker-entrypoint.sh (hunk 3 above)
Comment (sdake) [reply docker/build-tools/docker-entrypoint.sh]: resolved.

File: docker/build-tools/docker-entrypoint.sh (hunk 4 above)
Comment (sdake) [reply docker/build-tools/docker-entrypoint.sh]: resolved.

File: docker/build-tools/Dockerfile (hunk 1 above)
Comment (sdake) [reply docker/build-tools/Dockerfile]: addressed.

File: docker/build-tools/Dockerfile [hunk 5]
```diff
@@ -379,13 +390,18 @@ COPY --from=python_context /usr/local/bin /usr/local/bin
 COPY --from=python_context /usr/local/lib /usr/local/lib
 COPY --from=python_context /usr/local/google-cloud-sdk /usr/local/google-cloud-sdk
 
+# su-exec is used in place of complex sudo setup operations
+RUN chmod u+sx /usr/bin/su-exec
```
Comment (geeknoid) [inline docker/build-tools/Dockerfile]: Shouldn't this be done above, after line 163?

Comment (sdake) [reply docker/build-tools/Dockerfile]: no - it doesn't work unless it is in this final section.  it appears suid bits are not copied across COPY sections.

**Comment by geeknoid** [comment]
Just remove the do-not-merge label when ready...

**Comment by sdake** [comment]
i'm ready. :)

**Pull request merged**